### PR TITLE
Doc updates

### PR DESCRIPTION
**Pull request by jr2730**
- added support for generating API markdown file
  - added script to auto-install required support
  - added script to regenerate API markdown file
- added current instance of API markdown file
- added new README file with notes

### CODE REVIEW

File: ansible/dimensiondata/dimensiondata_backup_client.py [hunk 1]
```diff
@@ -14,7 +14,7 @@
 
 DOCUMENTATION = '''
 ---
-module: dimensiondata_backup
+module: dimensiondata_backup client
```
Comment (aimonb) [inline ansible/dimensiondata/dimensiondata_backup_client.py:17]: Why did you add the next 'client' here? That confuses the module name IMO.

File: ansible/dimensiondata/dimensiondata_backup.py [hunk 2]
```diff
@@ -38,8 +38,11 @@
   region:
     description:
       - The target region.
-    choices: %s
-    default: na
+    choices:
+      - Regions are defined in Apache libcloud project file = libcloud/common/dimensiondata.py 
+      - See https://libcloud.readthedocs.io/en/latest/compute/drivers/dimensiondata.html
+      - Note: Values avail in array dd_regions(). 
+    default: na aka North America
```
Comment (aimonb) [inline ansible/dimensiondata/dimensiondata_backup.py:45]: This can confuse the reader on what the actual default value is. Please remove here and in other places. They can read link you added for more info.

File: docs/README_DD_api.md [hunk 3]
```diff
@@ -0,0 +1,475 @@
...
+* See official Ansible docs
+
+---
+### Modules
+
+  * [dimensiondata_backup - enable or disable backups for a host.](#dimensiondata_backup)
+  * [dimensiondata_backup client - add/delete backup client for a host](#dimensiondata_backup client)
+  * [dimensiondata_compute - create, terminate, start or stop an server in dimensiondata](#dimensiondata_compute)
+  * [dimensiondata_disk - add/remove/modify disks](#dimensiondata_disk)
+  * [dimensiondata_get_unallocated_public_ips - get specified number of free addresses; provision to reach requested number.](#dimensiondata_get_unallocated_public_ips)
```
Comment (aimonb) [inline docs/README_DD_api.md]: You are missing some of the modules here. Will you add them later?

File: ansible/dimensiondata/dimensiondata_backup.py (hunk 2 above)
Comment (jr2730) [inline ansible/dimensiondata/dimensiondata_backup.py:45]: na seems to me be a very confusing value for a default value.

Makes is seem like NOT APPLICABLE.

Which is why I got more specific

File: docs/README_DD_api.md [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+ e
+ d
+  
+ n
+ u
+ m
+ b
+ e
+ r
+ .
```
Comment (aimonb) [inline docs/README_DD_api.md]: Whats up with the one char per line?

File: docs/README_DD_api.md (hunk 3 above)
Comment (jr2730) [inline docs/README_DD_api.md]: These are all that auto-generated.
Didn't realize some where missing.
Will check.

**Comment by aimonb** [comment]
Please see inline comments.

File: docs/README_DD_api.md (hunk 4 above)
Comment (jr2730) [inline docs/README_DD_api.md]: Bizarre.

File: ansible/dimensiondata/dimensiondata_backup_client.py (hunk 1 above)
Comment (jr2730) [inline ansible/dimensiondata/dimensiondata_backup_client.py:17]: This matches the name of the actual module (backup vs backup client).
Shouldn't it match the filename

File: ansible/dimensiondata/dimensiondata_backup.py (hunk 2 above)
Comment (aimonb) [inline ansible/dimensiondata/dimensiondata_backup.py:45]: how about adding note in () instead of aka?

Comment (jr2730) [inline ansible/dimensiondata/dimensiondata_backup.py:45]: Compromise accepted!

Comment (aimonb) [inline ansible/dimensiondata/dimensiondata_backup.py:45]: Please single quote 'na' also

File: docs/README_DD_api.md (hunk 3 above)
Comment (aimonb) [inline docs/README_DD_api.md]: Ah! thanks.

File: docs/README_DD_api.md (hunk 4 above)
Comment (aimonb) [inline docs/README_DD_api.md]: Ah right auto-generated.. but strange.. see what you can find.
